### PR TITLE
chore: refine the switchover action of apecloud-mysql and postgresql

### PR DESCRIPTION
**Pull request by leon-inf**

### CODE REVIEW

File: addons/postgresql/templates/componentdefinition-12.yaml [hunk 1]
```diff
@@ -138,22 +144,18 @@ spec:
...
-            - curl -s http://$(KB_REPLICATION_PRIMARY_POD_FQDN):8008/switchover -XPOST -d '{"leader":"$(KB_REPLICATION_PRIMARY_POD_NAME)"}'
+      exec:
+        image: {{ include "postgresql.image.major12.minor150" . }}
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -ex
+          if [ -z "$KB_SWITCHOVER_CANDIDATE_FQDN" ]; then
+              curl -s http://$(REPLICATION_PRIMARY_POD_FQDN):8008/switchover -XPOST -d '{"leader":"$(REPLICATION_PRIMARY_POD_NAME)"}'
```
Comment (nayutah) [inline addons/postgresql/templates/componentdefinition-12.yaml:155]: add curl timeout and retry count like 'curl --retry 3 --connect-timeout 5 --max-time 10 '

File: addons/postgresql/templates/componentdefinition-14.yaml [hunk 2]
```diff
@@ -138,22 +144,18 @@ spec:
...
-            - curl -s http://$(KB_REPLICATION_PRIMARY_POD_FQDN):8008/switchover -XPOST -d '{"leader":"$(KB_REPLICATION_PRIMARY_POD_NAME)"}'
+      exec:
+        image: {{ include "postgresql.image.major14.minor080" . }}
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -ex
+          if [ -z "$KB_SWITCHOVER_CANDIDATE_FQDN" ]; then
+              curl -s http://$(REPLICATION_PRIMARY_POD_FQDN):8008/switchover -XPOST -d '{"leader":"$(REPLICATION_PRIMARY_POD_NAME)"}'
```
Comment (nayutah) [inline addons/postgresql/templates/componentdefinition-14.yaml:155]: ditto

File: addons/postgresql/templates/componentdefinition-15.yaml [hunk 3]
```diff
@@ -140,22 +146,18 @@ spec:
...
-            - curl -s http://$(KB_REPLICATION_PRIMARY_POD_FQDN):8008/switchover -XPOST -d '{"leader":"$(KB_REPLICATION_PRIMARY_POD_NAME)"}'
+      exec:
+        image: {{ include "postgresql.image.major15.minor070" . }}
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -ex
+          if [ -z "$KB_SWITCHOVER_CANDIDATE_FQDN" ]; then
+              curl -s http://$(REPLICATION_PRIMARY_POD_FQDN):8008/switchover -XPOST -d '{"leader":"$(REPLICATION_PRIMARY_POD_NAME)"}'
```
Comment (nayutah) [inline addons/postgresql/templates/componentdefinition-15.yaml:157]: ditto

**Comment by Y-Rookie** [comment]
outdated

**Pull request closed without merging**